### ISSUE TITLE
C#: Calling signal callback with wrong parameter types raises confusing "Method not found" error.

### ISSUE THREAD

**Issue by thomasDelaporte** [godotengine/godot#35910]
[Networking.zip](https://github.com/godotengine/godot/files/4155965/Networking.zip)

**Godot version:** 3.1.2

**OS/device including version:** Windows 10

**Issue description:**
When the the server (specific godot project) received an event (signal) with the name "network_peer_packet", the system should fire the method "ReceivedPacket" in the class Server.cs.

But instead it doesn't found it.

`ERROR: emit_signal: Error calling method from signal 'network_peer_packet': 'Node2D(MinimalServer.cs)::ReceivedPacket': Method not found.
   At: core/object.cpp:1236`

**Steps to reproduce:**
> Create a server with NetworkedMultiplayerENet
> Connect signal "network_peer_packet" to a method 

> Create a client
> SendByte to the server

**Minimal reproduction project:**

Server side :
```  
public class MinimalServer : Node {
        
        public readonly int PORT = 4041;
        public readonly int MAX_USERS = 2;

        private NetworkedMultiplayerENet host;
     
        public override void _Ready() {
          
            this.host = new NetworkedMultiplayerENet();
            Error error = host.CreateServer(PORT, MAX_USERS);
          
            if (error != Error.Ok) {
                GD.PrintErr(error); return;
            }
          
            this.GetTree().SetNetworkPeer(host);

            // HERE: Bind
            GetTree().Multiplayer.Connect("network_peer_packet", this, nameof(ReceivedPacket));
        }
     
        public void ReceivedPacket() {
            GD.Print("test");
        }
    }
``` 

Client side :
```
public class MinimalClient : Node {
        
        private readonly int PORT = 4041;
        private readonly string IP = "127.0.0.1";
    
        public NetworkedMultiplayerENet client;

        public override void _Ready() {
            
            this.client = new NetworkedMultiplayerENet();
            Error error = client.CreateClient(IP, PORT);
        
            if (error != Error.Ok) {
                GD.PrintErr(error); return;
            }
        
            this.GetTree().SetNetworkPeer(client);
            GD.Print("Connected to server " + IP + " and port " + PORT);
        }
        
        public override void _Input(InputEvent @event) {
        
            if(this.client == null)
                return;
        
            if (!(@event is InputEventKey))
                return;
        
            InputEventKey e = (InputEventKey) @event;
            if (e.IsPressed() && e.Scancode == (int) KeyList.Enter) {

                String payload = "Hello world";
            
                GD.Print("[CLIENT] Send tchat message with payload : " + payload);
                GetTree().Multiplayer.SendBytes(Encoding.UTF8.GetBytes(payload));
            }
        }
    }
```

**Action**: PRESS ENTER to send "Hello world" to server.

+ You can find a ZIP with the server and the client project.
+ In the client project, the UI is not important ( could be deleted )

[Networking.zip](https://github.com/godotengine/godot/files/4155967/Networking.zip)

**Comment by andy-noisyduck** [comment]
The Mono error for this isn't great, but it can't find the method because the expected signatures don't match.

From the [3.2 docs](https://docs.godotengine.org/en/3.2/classes/class_multiplayerapi.html):
`
network_peer_packet ( int id, PoolByteArray packet )
`

Which in c# would translate to `(int, byte[])` accordingly. If you update your `ReceivedPacket()` handler to match your example project will no longer error.

**Comment by thomasDelaporte** [comment]
Great, would be nice to have a better error.

Thanks.

**Comment by akien-mga** [comment]
Reopening, IMO the error message could likely be improved to properly handle this kind of situation.

**Comment by nobuyukinyuu** [comment]
Curious to know how this error effects binds.  Having to pass in one as `Godot.Collections.Array(new object[]{ doubleValue })` here and having (double x) in the target method doesn't seem to be working.  

Connect code:
```
    public void AttachNoteToSignal(Note note, Node signalSource)
    {
        signalSource.Connect("NoteOff", note as Note, "_on_ReleaseNote", new Godot.Collections.Array(new object[]{patch.GetReleaseTime()}) );
    }
```

Error message (It seems to be looking in the base class rather than the derived one...):
```
ERROR: emit_signal: Error calling method from signal 'NoteOff': 'Node::_on_ReleaseNote': Method not found..
```

**Comment by andy-noisyduck** [comment]
I think you might have something else going on. I just did a quick test with extra params in the bindings, and Godot seems happy to call any method where the number of arguments matches - the types seemed to be ignored.

FWIW, your example has a double, inside a C# array, inside a Godot array. Was that intended?

**Comment by nobuyukinyuu** [comment]
> FWIW, your example has a double, inside a C# array, inside a Godot array. Was that intended?

Yeah that was intended, since the binds argument requires a Godot array as its type and needs to be initialized with an IEnumerable (a ` new object[]` doesn't seem to automatically cast to this type despite a number of examples online implying otherwise).  

I should note that the signal is being emitted from a GDScript, so perhaps there's a difference in marshaling there?  

Actually, nevermind, the problem seemed to go away when adding an overload of the method with no parameters (?!)....  Guess this warrants opening a new issue!
